c/src/ml-api-service-training-offloading.h

```diff
@@ -0,0 +1,125 @@
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to open the model.
+ * @retval #ML_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ */
+  int ml_service_training_offloading_create (ml_service_s * mls, JsonObject * object);
+
+/**
+ * @brief Set path in ml-service training offloading handle.
+ * @note This is not official and public API but experimental API.
+ * @param[in] mls ml-service handle created by ml_service_new().
+ * @param[in] path Readable and writable path set by the app.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Given parameter is invalid.
+ */
+  int ml_service_training_offloading_set_path (ml_service_s *mls, const gchar * path);
+
+/**
+ * @brief Start ml training offloading service.
+ * @remarks The @a handle should be destroyed using ml_service_destroy().
+ * @param[in] mls ml-service handle created by ml_service_new().
+ * @return @c 0 on Success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to launch the pipeline.
+ * @retval #ML_ERROR_TRY_AGAIN The pipeline is not ready yet.
+ * @retval #ML_ERROR_PERMISSION_DENIED The application does not have the privilege to access to the storage.
+ */
+  int ml_service_training_offloading_start (ml_service_s * mls);
+
+/**
+ * @brief Stop ml training offloading service.
+ * @remarks The @a handle should be destroyed using ml_service_destroy().
+ * @param[in] mls ml-service handle created by ml_service_new().
+ * @return @c 0 on Success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to launch the pipeline.
+ * @retval #ML_ERROR_TRY_AGAIN The pipeline is not ready yet.
+ * @retval #ML_ERROR_PERMISSION_DENIED The application does not have the privilege to access to the storage.
+ */
+  int ml_service_training_offloading_stop (ml_service_s * mls);
+
+/**
+ * @brief Request all services to ml-service offloading.
+ * @param[in] mls ml-service handle created by ml_service_new().
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Given parameter is invalid.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Given parameter is invalid.
```

Comment:
```suggestion
 * @return @c 0 on success. Otherwise, a negative error value.
 * @retval #ML_ERROR_NONE Successful.
 * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
 * @retval #ML_ERROR_INVALID_PARAMETER Given parameter is invalid.
```

Reply:
Thank you.